packages/bpk-component-nudger/src/BpkNudger.js

```diff
@@ -44,82 +47,34 @@ const BpkNudger = props => {
     decreaseButtonLabel,
     buttonType,
   } = props;
-  const classNames = [getClassName('bpk-nudger')];
-  if (className) {
-    classNames.push(className);
-  }
 
-  const adjustedValue = Math.floor(clamp(value, min, max));
-  const decreaseDisabled = adjustedValue <= min;
-  const increaseDisabled = adjustedValue >= max;
-
-  const minusIconClassNames = [getClassName('bpk-nudger__icon')];
-  if (decreaseDisabled) {
-    minusIconClassNames.push(getClassName('bpk-nudger__icon--disabled'));
-  }
-  const plusIconClassNames = [getClassName('bpk-nudger__icon')];
-  if (increaseDisabled) {
-    plusIconClassNames.push(getClassName('bpk-nudger__icon--disabled'));
-  }
-
-  const inputStyles = [getClassName('bpk-nudger__input')];
-  if (buttonType === 'outline') {
-    inputStyles.push(getClassName('bpk-nudger__input--outline'));
-  }
+  const compareValues = (a: number, b: number): number => a - b;
+  const incrementValue = (a: number): number => a + 1;
+  const decrementValue = (a: number): number => a - 1;
+  const formatValue = (a: number): string => a.toString();
 
   return (
-    <div className={classNames.join(' ')}>
-      <BpkButton
-        secondary={buttonType === 'secondary'}
-        outline={buttonType === 'outline'}
-        iconOnly
-        onClick={() => onChange(clamp(adjustedValue - 1, min, max))}
-        disabled={decreaseDisabled}
-        title={decreaseButtonLabel}
-        aria-controls={id}
-        className={getClassName('bpk-nudger__button')}
-      >
-        <AlignedMinusIcon className={minusIconClassNames.join(' ')} />
-      </BpkButton>
-      <input
-        type="text"
-        aria-live="assertive"
-        readOnly
-        value={adjustedValue}
-        id={id}
-        className={inputStyles.join(' ')}
-      />
-      <BpkButton
-        secondary={buttonType === 'secondary'}
-        outline={buttonType === 'outline'}
-        iconOnly
-        onClick={() => onChange(clamp(adjustedValue + 1, min, max))}
-        disabled={increaseDisabled}
-        title={increaseButtonLabel}
-        aria-controls={id}
-        className={getClassName('bpk-nudger__button')}
-      >
-        <AlignedPlusIcon className={plusIconClassNames.join(' ')} />
-      </BpkButton>
-    </div>
+    <BpkConfigurableNudger
+      id={id}
+      min={min}
+      max={max}
+      value={value}
+      onChange={onChange}
+      className={className}
+      inputClassName={getClassName('bpk-nudger__input--numeric')}
+      increaseButtonLabel={increaseButtonLabel}
+      decreaseButtonLabel={decreaseButtonLabel}
+      buttonType={buttonType}
+      compareValues={compareValues}
+      incrementValue={incrementValue}
+      decrementValue={decrementValue}
+      formatValue={formatValue}
```

Comment:
If you're not doing any logic on `id`, `min`, `max` and so on, you can just pass them through with `...rest` to save some code